coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/utils/CoralToTrinoSqlCallTransformersUtil.java

```diff
@@ -0,0 +1,206 @@
+                + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+            null, null));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(new OperatorBasedSqlCallTransformer(hiveToCoralSqlOperator("datediff"), 2,
+        "date_diff",
+        "[{\"value\": 'day'}, {\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 2}]}]}, "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+  }
+
+  private static void addLinkedInFunctionTransformers() {
+    // Most "com.linkedin..." UDFs follow convention of having UDF names mapped from camel-cased name to snake-cased name.
+    // For example: For class name IsGuestMemberId, the conventional udf name would be is_guest_member_id.
+    // While this convention fits most UDFs it doesn't fit all. With the following mapping we override the conventional
+    // UDF name mapping behavior to a hardcoded one.
+    // For example instead of UserAgentParser getting mapped to user_agent_parser, we mapped it here to useragentparser
+    Set<String> linkedInFunctionSignatureSet = new HashSet<>();
+    addLinkedInFunctionTransformer("com.linkedin.dali.udf.watbotcrawlerlookup.hive.WATBotCrawlerLookup", 3,
+        "wat_bot_crawler_lookup", linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.Ip2Str", 1, "ip2str",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.Ip2Str", 3, "ip2str",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.UserAgentParser", 2, "useragentparser",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.lookup.hive.BrowserLookup", 3, "browserlookup",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.jobs.udf.hive.ConvertIndustryCode", 1, "converttoindustryv1",
+        linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper", 1,
+        "urn_extractor", linkedInFunctionSignatureSet);
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper", 1, "urn_extractor",
+        linkedInFunctionSignatureSet);
+
+    addLinkedInFunctionTransformerFromHiveRegistry(DEFAULT_SQL_CALL_TRANSFORMER_LIST, linkedInFunctionSignatureSet);
+  }
+
+  private static void addLinkedInFunctionTransformer(String linkedInFuncName, int numOperands, String trinoFuncName,
+      Set<String> linkedInFunctionSignatureSet) {
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(new OperatorBasedSqlCallTransformer(
+        linkedInFunctionToCoralSqlOperator(linkedInFuncName), numOperands, trinoFuncName));
+    linkedInFunctionSignatureSet.add(linkedInFuncName + "_" + numOperands);
```

Comment:
All the keys in `HIVE_FUNCTION_REGISTRY` (`StaticHiveFunctionRegistry`) are lowercased, so `linkedInFuncName` should be `linkedInFuncName.toLowerCase()`, otherwise the following `linkedInFunctionSignatureSet.contains(hiveFunctionName + "_" + i)` will always to false.

Reply:
Good catch, the original code in `CalciteTrinoUDFMap` has this problem as shown in https://github.com/linkedin/coral/blob/7c23b8d5c04528d982d6103a54a80573e3832cf1/coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CalciteTrinoUDFMap.java#L124
I have fixed it here.